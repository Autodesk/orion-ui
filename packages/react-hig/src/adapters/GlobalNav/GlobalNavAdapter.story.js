/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

 */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { select, boolean, text } from '@storybook/addon-knobs';

import GlobalNav from './GlobalNavAdapter';

const SideNav = GlobalNav.SideNav;
const Search = GlobalNav.SideNav.Search;
const SectionList = GlobalNav.SideNav.SectionList;
const Section = GlobalNav.SideNav.SectionList.Section;
const Collapse = GlobalNav.SideNav.SectionList.Section.SectionCollapse;
const Group = GlobalNav.SideNav.SectionList.Section.Group;
const LinkList = GlobalNav.SideNav.LinkList;
const Link = GlobalNav.SideNav.LinkList.Link;
const Module = GlobalNav.SideNav.SectionList.Section.Group.Module;
const ModuleCollapse = GlobalNav.SideNav.SectionList.Section.Group.Module.ModuleCollapse;
const Submodule = GlobalNav.SideNav.SectionList.Section.Group.Module.Submodule;
const TopNav = GlobalNav.TopNav;
const Profile = TopNav.Profile;
const Shortcut = TopNav.Shortcut;
const Help = TopNav.Help;
const TopNavSearch = TopNav.Search;
const ProjectAccountSwitcher = GlobalNav.TopNav.ProjectAccountSwitcher;
const Account = GlobalNav.TopNav.ProjectAccountSwitcher.Account;
const Project = GlobalNav.TopNav.ProjectAccountSwitcher.Project;
const SubNav = GlobalNav.SubNav;
const Tabs = GlobalNav.SubNav.Tabs;
const Tab = GlobalNav.SubNav.Tabs.Tab;
const Slot = GlobalNav.Slot;

const links = [
  { title: 'Autodesk Main', url: 'http://www.autodesk.com' },
  {
    title: 'AutoCAD',
    url: 'https://www.autodesk.com/products/autocad/overview'
  },
  { title: 'Maya', url: 'https://www.autodesk.com/products/maya/overview' }
];
const LONG_COPY = (
  <div>
    <p>
      Raw denim flexitarian green juice kinfolk. Umami hammock trust fund everyday carry, woke wolf viral sriracha austin.
      Fingerstache affogato messenger bag salvia, cray iPhone bushwick blue bottle marfa gentrify dreamcatcher pop-up.
      Slow-carb etsy enamel pin cronut, raclette post-ironic hashtag. Hoodie dreamcatcher enamel pin lumbersexual before
      they sold out, authentic selvage tumblr vinyl. Hot chicken chillwave coloring book fixie vice venmo echo park
      portland. Tote bag master cleanse cronut banjo banh mi pitchfork, celiac photo booth.
    </p>

    <p>
      Next level deep v roof party, jianbing pok pok pug butcher vape farm-to-table kombucha. Yr snackwave VHS, wolf
      poutine actually woke poke flexitarian paleo food truck DIY kale chips viral yuccie. Cornhole tattooed vaporware
      affogato, gentrify mlkshk portland organic. Swag try-hard cronut hashtag, etsy bespoke chia banjo messenger bag.
      Mustache umami godard man braid cronut yuccie. YOLO vaporware franzen, gochujang typewriter mixtape brunch salvia
      paleo lyft. Four dollar toast tumblr mustache thundercats single-origin coffee, freegan flexitarian cold-pressed
      beard roof party VHS venmo af ugh bushwick.
    </p>

    <p>
      Ethical man bun shoreditch chambray farm-to-table. Celiac readymade tote bag shabby chic chia, franzen poke meggings
      pop-up williamsburg VHS. Flexitarian cardigan blog yuccie activated charcoal farm-to-table. Iceland activated
      charcoal aesthetic, chambray offal snackwave austin four loko stumptown roof party humblebrag. Food truck selfies
      vexillologist, subway tile direct trade fingerstache tofu biodiesel four dollar toast. Literally squid tattooed jean
      shorts, raw denim echo park blog mustache skateboard seitan XOXO lo-fi cray hammock shabby chic. Live-edge
      sustainable pug, pinterest celiac kinfolk wayfarers narwhal art party vegan +1.
    </p>

    <p>
      Helvetica DIY portland, sriracha selvage wolf beard plaid quinoa vegan flexitarian poke. Semiotics fam scenester,
      plaid synth listicle flexitarian subway tile. Cred everyday carry pop-up, forage viral tbh tilde mustache roof party
      leggings gastropub unicorn synth. Brunch hoodie williamsburg, selfies plaid jianbing deep v lomo truffaut meggings
      dreamcatcher kitsch. Mustache deep v fap hashtag, freegan quinoa blue bottle green juice helvetica pickled cronut.
      Photo booth 90's microdosing vexillologist roof party stumptown. Viral hella pitchfork, aesthetic intelligentsia
      godard hoodie plaid migas iPhone mlkshk tumeric keffiyeh.
    </p>

    <p>
      Banh mi pug fanny pack heirloom portland. Humblebrag selvage forage vape organic beard bicycle rights, fam gentrify
      live-edge. Vape artisan truffaut, celiac prism XOXO drinking vinegar pour-over vinyl chambray lyft venmo. Tattooed
      edison bulb air plant hot chicken meggings cold-pressed gastropub, tbh tilde hoodie photo booth leggings. Etsy
      post-ironic fap you probably haven't heard of them, helvetica synth kinfolk listicle mixtape keytar cold-pressed.
      90's put a bird on it yr godard prism kogi, live-edge enamel pin hell of photo booth. Raclette portland humblebrag
      synth af, fashion axe fingerstache ethical snackwave post-ironic cray.
    </p>

    <p>
      Messenger bag listicle skateboard normcore. Roof party food truck authentic dreamcatcher lumbersexual ramps. Man bun
      drinking vinegar mixtape, disrupt sartorial cray lumbersexual bicycle rights fam brunch celiac lyft. Hella edison
      bulb poke thundercats taxidermy twee. Lomo biodiesel la croix, pitchfork normcore prism godard coloring book
      cornhole locavore helvetica. Everyday carry distillery tousled, bespoke irony chartreuse lumbersexual fixie. Unicorn
      asymmetrical tacos poke, tote bag disrupt sriracha coloring book.
    </p>

    <p>
      Humblebrag narwhal hammock, cornhole biodiesel lomo vegan twee migas enamel pin iPhone vaporware chicharrones vape
      wolf. Godard XOXO deep v tbh irony, church-key seitan fixie post-ironic asymmetrical literally put a bird on it.
      Hell of helvetica cornhole lomo forage. Affogato activated charcoal ugh, +1 vape poutine poke artisan listicle
      before they sold out brunch prism VHS cronut. Craft beer PBR&B vice, synth XOXO green juice bitters narwhal
      chicharrones pinterest microdosing intelligentsia wayfarers scenester schlitz. Stumptown synth bicycle rights ugh
      poke, photo booth quinoa cronut pickled meggings tumeric. Yr wayfarers mustache pitchfork, art party bitters craft
      beer single-origin coffee.
    </p>
  </div>
);

import logo from '../../images/bim-logo.png';
import profileImage from '../../images/profileImage.png';
import project1 from '../../images/project-1.png';
import project2 from '../../images/project-2.png';
import project3 from '../../images/project-3.png';
import project4 from '../../images/project-4.png';

storiesOf('GlobalNav', module)
  .addWithInfo('default', <div><p>Global Nav basic usage</p></div>, () => {
    return (
      <GlobalNav>
        <SideNav>
          <SectionList>
            <Section headerLabel="Project" headerName="ThunderStorm">
              <Group>
                <Module icon="insight" title="Insight" key={'Insight'}>
                  <Submodule
                    title="Overview"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
              </Group>
            </Section>
            <Section headerLabel="Account" headerName="Global Construction">
              <Group>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                >
                  <Submodule
                    title="Checklists Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Issues Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                />
              </Group>
            </Section>
          </SectionList>
          <Search placeholder="Find module or submodule" />
        </SideNav>
        <TopNav logo={logo}>
          <ProjectAccountSwitcher
            open={boolean('Project Account Switcher open:', false)}
            onProjectChange={action('Project activated')}
            onAccountChange={action('Account activated')}
          >
            <Account image="" label="DPI Construction" />
            <Project image={project4} label="Oakland Medical Center" />
            <Project image={project1} label="Stanford hospital" />
          </ProjectAccountSwitcher>
          <Shortcut icon="gear" title="GEAR" link="#" />
          <Help title="HELLLP MEEEE!!!!" link="#`" />
          <Profile
            image={profileImage}
            name="Jane Designer"
            email="jane.designer@example.com"
            onProfileSettingsClick={action('profile settings clicked')}
            onSignOutClick={action('sign out clicked')}
          />
        </TopNav>
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('with SideNavOpen Prop passed in', ``, () => {
    const sideNavOpen = boolean('sideNavOpen', true);
    return (
      <GlobalNav sideNavOpen={sideNavOpen}>
        <SideNav>
          <SectionList>
            <Section headerLabel="Project" headerName="ThunderStorm">
              <Group>
                <Module icon="insight" title="Insight" key={'Insight'}>
                  <Submodule
                    title="Overview"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
              </Group>
            </Section>
            <Section headerLabel="Account" headerName="Global Construction">
              <Group>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                >
                  <Submodule
                    title="Checklists Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Issues Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                />
              </Group>
            </Section>
          </SectionList>

          <LinkList>
            {links.map((link, i) => {
              return (
                <Link
                  title={text(`Link ${i + 1} title`, link.title)}
                  link={link.url}
                  key={i}
                />
              );
            })}
          </LinkList>

          <Search placeholder="Find module or submodule" />
        </SideNav>
        <TopNav logo={logo} />
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('w profileFlyoutOpen', ``, () => {
    const profileFlyoutOpenKnob = boolean('profileFlyoutOpen', true);
    return (
      <GlobalNav>
        <SideNav />
        <TopNav logo={logo}>
          <Profile
            image={profileImage}
            name="Jane Designer"
            open="true"
            email="jane.designer@example.com"
            onProfileSettingsClick={action('profile settings clicked')}
            onSignOutClick={action('sign out clicked')}
          />
        </TopNav>
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('w projectAccountSwitcher Single Account', ``, () => {
    const isOpen = boolean('isOpen', true);
    return (
      <GlobalNav>
        <SideNav />
        <TopNav logo={logo}>
          <ProjectAccountSwitcher
            activeLabel="Oakland Medical Center"
            activeImage={project1}
            activeType="account"
            hideProjectAccountFlyout={true}
          />
        </TopNav>
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('w sideNavOpen with query', ``, () => {
    const sideNavOpen = boolean('sideNavOpen', true);
    const query = text('query', 'Field');
    return (
      <GlobalNav sideNavOpen={sideNavOpen}>
        <SideNav>
          <SectionList>
            <Section headerLabel="Project" headerName="ThunderStorm">
              <Group>
                <Module icon="insight" title="Insight" key={'Insight'}>
                  <Submodule
                    title="Overview"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
              </Group>
            </Section>
            <Section headerLabel="Account" headerName="Global Construction">
              <Group>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                >
                  <Submodule
                    title="Checklists Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Issues Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                />
              </Group>
            </Section>
          </SectionList>
          <Search placeholder="Find module or submodule" query={query} />
        </SideNav>
        <TopNav logo={logo} />
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('w sideNavOpen without query', ``, () => {
    const sideNavOpen = boolean('sideNavOpen', true);
    return (
      <GlobalNav sideNavOpen={sideNavOpen}>
        <SideNav>
          <SectionList>
            <Section headerLabel="Project" headerName="ThunderStorm">
              <Group>
                <Module icon="insight" title="Insight" key={'Insight'}>
                  <Submodule
                    title="Overview"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
              </Group>
            </Section>
            <Section headerLabel="Account" headerName="Global Construction">
              <Group>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                >
                  <Submodule
                    title="Checklists Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Issues Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                />
              </Group>
            </Section>
          </SectionList>
          <Search placeholder="Find module or submodule" />
        </SideNav>
        <TopNav logo={logo} />
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('SideNav open w/ custom hover/click Links', ``, () => {
    return (
      <GlobalNav sideNavOpen={true}>
        <SideNav>
          <SectionList>
            <Section headerLabel="Project" headerName="ThunderStorm">
              <Group>
                <Module icon="insight" title="Insight" key={'Insight'}>
                  <Submodule
                    title="Overview"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Extreme Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Calculated risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
              </Group>
            </Section>
          </SectionList>
          <LinkList>
            <Link
              title="Hello Kitty"
              link="#"
              onClick={action('I was clicked')}
            />
            <Link
              title="Dear Daniel"
              link="http://www.sanrio.com"
              onHover={action("I'm being hovered")}
            />
          </LinkList>
          <Search placeholder="Find module or submodule" />
        </SideNav>
        <TopNav logo={logo} />
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('? no children', ``, () => {
    const sideNavOpen = boolean('sideNavOpen', true);
    return <GlobalNav sideNavOpen={sideNavOpen} />;
  })
  .addWithInfo('? no SideNav', ``, () => {
    const sideNavOpen = boolean('sideNavOpen', true);
    return (
      <GlobalNav sideNavOpen={sideNavOpen}>
        <TopNav logo={logo} />
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('? no TopNav', ``, () => {
    const sideNavOpen = boolean('sideNavOpen', true);
    return (
      <GlobalNav sideNavOpen={sideNavOpen}>
        <SideNav>
          <SectionList>
            <Section headerLabel="Project" headerName="ThunderStorm">
              <Collapse onClick={action('clicked')} isCollapsed={true} />
              <Group>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                >
                  <Submodule
                    title="Checklists Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
              </Group>
            </Section>
            <Section
              headerLabel="Contractor"
              headerName="Oakland Medical Center"
            >
              <Collapse onClick={action('clicked')} isCollapsed={false} />
              <Group>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                >
                  <Submodule
                    title="Checklists Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
              </Group>
            </Section>
          </SectionList>
        </SideNav>
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('? no SubNav', ``, () => {
    const sideNavOpen = boolean('sideNavOpen', false);
    return (
      <GlobalNav sideNavOpen={sideNavOpen}>
        <SideNav>
          <SectionList>
            <Section headerLabel="Project" headerName="ThunderStorm">
              <Collapse onClick={action('clicked')} isCollapsed={true} />
              <Group>
                <Module
                  icon="project-management"
                  title="Item 1"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={true}
                >
                  <Submodule
                    title="Item 2"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
              </Group>
            </Section>
          </SectionList>
        </SideNav>
        <TopNav logo={logo} />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('empty Group in SideNav', ``, () => {
    const sideNavOpen = boolean('sideNavOpen', true);
    return (
      <GlobalNav sideNavOpen={sideNavOpen}>
        <SideNav>
          <SectionList>
            <Section headerLabel="Project" headerName="ThunderStorm">
              <Group>
                <Module icon="insight" title="Insight" key={'Insight'}>
                  <Submodule
                    title="Overview"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                  <Submodule
                    title="Risk"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
              </Group>
            </Section>
            <Section headerLabel="Account" headerName="Global Construction">
              <Group />
            </Section>
          </SectionList>
          <Search placeholder="Find module or submodule" />
        </SideNav>
        <TopNav logo={logo} />
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('empty Section in SideNav', ``, () => {
    const sideNavOpen = boolean('sideNavOpen', true);
    return (
      <GlobalNav sideNavOpen={sideNavOpen}>
        <SideNav>
          <SectionList>
            <Section headerLabel="Project" headerName="ThunderStorm">
              <Collapse onClick={action('clicked')} isCollapsed={true} />
              <Group>
                <Module
                  icon="field"
                  title="Field"
                  link="#"
                  onClick={action('clicked')}
                  submodulesClosed={false}
                >
                  <Submodule
                    title="Checklists Templates"
                    link="#"
                    onClick={action('clicked')}
                  />
                </Module>
              </Group>
            </Section>
            <Section
              headerLabel="Contractor"
              headerName="Oakland Medical Center"
            />
          </SectionList>
        </SideNav>
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger" />
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('? No Sections', ``, () => {
    const sideNavOpen = boolean('sideNavOpen', true);
    return (
      <GlobalNav sideNavOpen={sideNavOpen}>
        <SideNav />
        <TopNav logo={logo} />
        <Slot>
          {LONG_COPY}
        </Slot>
      </GlobalNav>
    );
  })
  .addWithInfo('w Tabs', ``, () => {
    const options = {
      Audience: 'Audience',
      Acquisition: 'Acquisition',
      Behavior: 'Behavior'
    };
    const activeTab = select('Active Tab', options, 'Audience');

    return (
      <GlobalNav>
        <TopNav logo={logo} />
        <SubNav moduleIndicatorName="Insight" moduleIndicatorIcon="hamburger">
          <Tabs>
            <Tab label="Audience" active={activeTab === 'Audience'} />
            <Tab label="Acquisition" active={activeTab === 'Acquisition'} />
            <Tab label="Behavior" active={activeTab === 'Behavior'} />
          </Tabs>
        </SubNav>
        <Slot>{LONG_COPY}</Slot>
      </GlobalNav>
    );
  });
