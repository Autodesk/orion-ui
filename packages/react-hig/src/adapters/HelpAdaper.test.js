/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

 */
import { mount } from 'enzyme';
import * as HIG from 'hig.web';
import React from 'react';

import GlobalNav from './GlobalNav/GlobalNavAdapter';
import TopNav from '../elements/components/GlobalNav/TopNav/TopNav';
import Help from './HelpAdapter';

const Context = props => {
  return (
    <GlobalNav>
      <TopNav>
        <Help title={props.title} link={props.link} />
      </TopNav>
    </GlobalNav>
  );
};

function setupReactContext(props) {
  const reactContainer = document.createElement('div');
  const reactWrapper = mount(<Context {...props} />, {
    attachTo: reactContainer
  });
  return { reactWrapper, reactContainer };
}

function createHigContext(props) {
  const higContainer = document.createElement('div');

  const higNav = new HIG.GlobalNav();
  higNav.mount(higContainer);

  const higTopNav = new higNav.partials.TopNav({});
  higNav.addTopNav(higTopNav);

  const help = new higTopNav.partials.Help(props);

  higTopNav.addHelp(help);

  return { higContainer, higItem: help };
}

describe('Help', () => {
  it('renders help ', () => {
    const props = {};
    const { reactWrapper, reactContainer } = setupReactContext(props);
    const { higContainer, higItem } = createHigContext(props);

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
    expect(reactContainer.firstElementChild.outerHTML).toEqual(
      higContainer.firstElementChild.outerHTML
    );
  });

  it('setting and updating props', () => {
    const defaults = {
      title: 'test',
      link: 'testLink'
    };

    const nextProps = {
      title: 'newTest',
      link: 'newTestLink'
    };

    const { higContainer, higItem } = createHigContext(defaults);
    const { reactWrapper, reactContainer } = setupReactContext(defaults);

    reactWrapper.setProps(nextProps);

    higItem.setTitle(nextProps.title);
    higItem.setLink(nextProps.link);

    expect(reactContainer.firstElementChild.outerHTML).toMatchSnapshot();
    expect(reactContainer.firstElementChild.outerHTML).toEqual(
      higContainer.firstElementChild.outerHTML
    );
  });
});
