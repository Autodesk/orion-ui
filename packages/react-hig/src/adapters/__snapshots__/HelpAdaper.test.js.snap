// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help renders help  1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
        <div class=\\"hig__global-nav__top-nav__hamburger\\">
            <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 30 30\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-92 -185)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><rect x=\\"92\\" y=\\"185\\" width=\\"30\\" height=\\"30\\"></rect><path d=\\"M100,195 L115,195 L115,196 L100,196 L100,195 Z M100,200 L115,200 L115,201 L100,201 L100,200 Z M100,205 L115,205 L115,206 L100,206 L100,205 Z\\" fill=\\"#34495E\\"></path></g></svg></div></div>
            <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><title>hig_icon_xclose</title><g fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect transform=\\"rotate(45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect transform=\\"rotate(-45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
        </div>
        <div class=\\"hig__global-nav__top-nav__logo\\">
            <a href=\\"#\\">
                <img src=\\"link\\">
            </a>
        </div>
    </div>
    <!--SEARCH-->
    <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
    <!--PROJECT_ACCOUNT_SWITCHER-->
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
    <div class=\\"hig__global-nav__top-nav__shortcut\\">
  <a href=\\"#\\" title=\\"\\" class=\\"hig__global-nav__top-nav__shortcut__link\\">
    <div class=\\"hig__global-nav__top-nav__shortcut__link__icon hig__global-nav__top-nav__shortcut__link__icon--help\\">
      <div class=\\"hig__icon\\"><svg width=\\"18\\" height=\\"18\\" viewBox=\\"0 0 18 18\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"help\\"><g fill=\\"#151E27\\" fill-rule=\\"evenodd\\" opacity=\\".573\\"><path d=\\"M9,17.125 C13.4873136,17.125 17.125,13.4873136 17.125,9 C17.125,4.51268641 13.4873136,0.875 9,0.875 C4.51268641,0.875 0.875,4.51268641 0.875,9 C0.875,13.4873136 4.51268641,17.125 9,17.125 Z M9,18 C4.02943725,18 0,13.9705627 0,9 C0,4.02943725 4.02943725,0 9,0 C13.9705627,0 18,4.02943725 18,9 C18,13.9705627 13.9705627,18 9,18 Z\\" fill-rule=\\"nonzero\\"></path><path d=\\"M8.016,10.678 L9.654,10.678 L9.654,10.328 C9.654,8.746 12.132,8.648 12.132,6.744 L12.132,6.324 C12.132,5.05 11.026,4 9.15,4 C7.75,4 6.826,4.476 6,5.134 L6.784,6.17 C7.554,5.638 8.1,5.344 8.996,5.344 C9.85,5.344 10.34,5.792 10.34,6.478 L10.34,6.814 C10.34,8.018 8.016,8.298 8.016,10.272 L8.016,10.678 Z M8.828,14.164 C9.458,14.164 9.934,13.674 9.934,13.058 C9.934,12.456 9.458,11.952 8.828,11.952 C8.226,11.952 7.736,12.456 7.736,13.058 C7.736,13.674 8.226,14.164 8.828,14.164 Z\\"></path></g></svg></div>
    </div>
  </a>
</div><!--SHORTCUT-->
    </div>
    <!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`Help setting and updating props 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
        <div class=\\"hig__global-nav__top-nav__hamburger\\">
            <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 30 30\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-92 -185)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><rect x=\\"92\\" y=\\"185\\" width=\\"30\\" height=\\"30\\"></rect><path d=\\"M100,195 L115,195 L115,196 L100,196 L100,195 Z M100,200 L115,200 L115,201 L100,201 L100,200 Z M100,205 L115,205 L115,206 L100,206 L100,205 Z\\" fill=\\"#34495E\\"></path></g></svg></div></div>
            <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><title>hig_icon_xclose</title><g fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect transform=\\"rotate(45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect transform=\\"rotate(-45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
        </div>
        <div class=\\"hig__global-nav__top-nav__logo\\">
            <a href=\\"#\\">
                <img src=\\"link\\">
            </a>
        </div>
    </div>
    <!--SEARCH-->
    <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
    <!--PROJECT_ACCOUNT_SWITCHER-->
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
    <div class=\\"hig__global-nav__top-nav__shortcut\\" title=\\"newTest\\" href=\\"newTestLink\\">newTest</div><!--SHORTCUT-->
    </div>
    <!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;
